### PR TITLE
Add User Data documentation

### PR DESCRIPTION
**Pull request by daxmc99**
Add documentation for "user-data" enhancement for EKS. This also includes a warning informing the user to be careful modifying these parameters.
https://github.com/rancher/rancher/issues/16300

### CODE REVIEW

**Comment by daxmc99** [comment]
Wait to merge until https://github.com/rancher/rancher/pull/17811 has been merged.

**Comment by daxmc99** [comment]
@nathan-jenan-rancher

File: content/rancher/v2.x/en/cluster-provisioning/hosted-kubernetes-clusters/eks/_index.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ If you chose this option, you must also choose a **VPC & Subnet** that allow you
     Custom AMI Override | If you want to use a custom [Amazon Machine Image](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html#creating-an-ami) (AMI), specify it here.
     Minimum ASG Size | The minimum number of instances that your cluster will scale to during low traffic, as controlled by [Amazon Auto Scaling](https://docs.aws.amazon.com/autoscaling/ec2/userguide/what-is-amazon-ec2-auto-scaling.html).
     Maximum ASG Size | The maximum number of instances that your cluster will scale to during high traffic, as controlled by [Amazon Auto Scaling](https://docs.aws.amazon.com/autoscaling/ec2/userguide/what-is-amazon-ec2-auto-scaling.html).
-
+    User Data | Custom commands can to be passed to perform auotmated configuration tasks <b> WARNING: </b> Modifying this may cause your nodes to be unable to join the cluster.
```
Comment (nathan-jenan-rancher) [inline content/rancher/v2.x/en/cluster-provisioning/hosted-kubernetes-clusters/eks/_index.md]: spelling mistake, should say `automated` not `auotmated`

**Comment by nathan-jenan-rancher** [comment]
LGTM

**Pull request merged**